frontend/src/metabase/containers/SaveQuestionModal.tsx

```diff
@@ -66,163 +69,192 @@ const isOverwriteMode = (
-                      />
-                      <FormCollectionPicker
-                        name="collection_id"
-                        title={t`Which collection should this go in?`}
-                      />
-                    </div>
-                  </CSSTransition>
-                )}
-              </TransitionGroup>
-              <FormFooter>
-                <FormErrorMessage inline />
-                <Button
-                  type="button"
-                  onClick={this.props.onClose}
-                >{t`Cancel`}</Button>
-                <FormSubmitButton title={t`Save`} primary />
-              </FormFooter>
-            </Form>
-          )}
-        </FormProvider>
-      </ModalContent>
+      <CreateCollectionModal
+        collectionId={openCollectionId}
+        onClose={() => setCreatingNewCollection(false)}
+        onCreate={(collection: Collection) => {
+          handleSubmit({ ...stagedValues, collection_id: collection.id });
+        }}
+      />
     );
   }
-}
+
+  return (
+    <ModalContent id="SaveQuestionModal" title={title} onClose={onClose}>
+      <FormProvider
+        initialValues={initialValues}
+        onSubmit={handleSubmit}
+        validationSchema={SAVE_QUESTION_SCHEMA}
+        enableReinitialize
+      >
+        {({ values, isValid }) => (
+          <Form>
+            {showSaveType && (
+              <FormRadio
+                name="saveType"
+                title={t`Replace or save as new?`}
+                options={[
+                  {
+                    name: t`Replace original question, "${originalQuestion?.displayName()}"`,
+                    value: "overwrite",
+                  },
+                  { name: t`Save as new question`, value: "create" },
+                ]}
+                vertical
+              />
+            )}
+            <TransitionGroup>
+              {values.saveType === "create" && (
+                <CSSTransition
+                  classNames="saveQuestionModalFields"
+                  timeout={{
+                    enter: 500,
+                    exit: 500,
+                  }}
+                >
+                  <div className="saveQuestionModalFields">
+                    <FormInput
+                      autoFocus
+                      name="name"
+                      title={t`Name`}
+                      placeholder={nameInputPlaceholder}
+                    />
+                    <FormTextArea
+                      name="description"
+                      title={t`Description`}
+                      placeholder={t`It's optional but oh, so helpful`}
+                    />
+                    <FormCollectionPicker
+                      onOpenCollectionChange={setOpenCollectionId}
+                      name="collection_id"
+                      title={t`Which collection should this go in?`}
+                    >
+                      <NewCollectionButton
+                        disabled={!isValid}
```

Comment:
We might want to add a tooltip that clarifies why is this button disabled because some users may want to select the collecton first and they won't understand why they can't create a new collection

Reply:
28d2cb99b7b9a58bf5699d0b896a6b2619f8fe97